share/functions/__fish_complete_ant_targets.fish

```diff
@@ -1,16 +1,68 @@
 function __fish_complete_ant_targets -d "Print list of targets from build.xml and imported files"
-    set -l buildfile "build.xml"
-    if test -f $buildfile
-        # show ant targets
-        __fish_filter_ant_targets $buildfile
-
-        # find files with buildfile
-        set files (sed -n "s/^.*<import[^>]* file=[\"']\([^\"']*\)[\"'].*\$/\1/p" < $buildfile)
-
-        # iterate through files and display their targets
-        for file in $files
+    function __filter_xml_start_tag -d "Filter xml start-tags in a buildfile"
+        set -l buildfile $argv[1] # full path to buildfile
+        set -l tag_pattern $argv[2] # regex pattern for tagname
+        # regex to filter start-tags ignoring newlines and '>' in attr values
+        # https://www.debuggex.com/r/wRgxHE1yTIgnjfNz
+        string join ' ' <$buildfile | string match -ar "<(?:$tag_pattern)(?:[^>\"']*?(?:(?:'[^']*?')|(?:\"[^\"]*\"))?)*?>"
+    end
+    function __filter_xml_attr_value -d "Filter xml attr value in a start-tag"
+        set -l tag $argv[1] # start-tag
+        set -l attr $argv[2] # attr name
+        # regex to filter attr values ignoring (single|double) quotes in attr values
+        # https://www.debuggex.com/r/x7lhtLJSP4msleik
+        string replace -rf "^.*$attr=((?:'(?:.*?)')|(?:\"(?:.*?)\")).*\$" '$1' $tag | string trim -c='"\''
+    end
+    function __get_buildfile -d "Get a buildfile that will be used by ant"
+        set -l tokens $argv # tokens from 'commandline -co'
+        set -l prev $tokens[1]
+        set -l buildfile "build.xml"
+        for token in $argv[2..-1]
+            switch $prev
+                case -buildfile -file -f
+                    set buildfile (eval echo $token)
+            end
+            set prev $token
+        end
+        # return last one
+        echo $buildfile
+    end
+    function __parse_ant_targets -d "Parse ant targets in the given build file"
+        set -l buildfile $argv[1] # full path to buildfile
+        set -l targets (__filter_xml_start_tag $buildfile 'target|extension-point')
+        for target in $targets
+            set -l target_name (__filter_xml_attr_value $target 'name')
+            if [ $status -eq 0 ]
```

Comment:
There's actually no need to do these as two separate lines, because `set` won't alter $status, so

```fish
if set -l target_name (__filter_xml_attr_value $target 'name')
```

effectively checks the status of `__filter_xml_attr_value $target 'name`.

Reply:
Thank you. I fixed.

Reply:
`set -l` in if condition does not work at v2.7.1. Is it a new feature?
```
$ fish -v
fish, version 2.7.1
$ function does_set-l_in_if_work
      if set -l var "foo"
          echo "var=$var"
      end
      echo "var=$var"
  end 
$ does_set-l_in_if_work
var=foo
var=
```

Reply:
It works at v3.0.0.
```
$ fish -v
fish, version 3.0.0
$ function does_set-l_in_if_work
            if set -l var "foo"
                    echo "var=$var"
            end
            echo "var=$var"
    end
$ does_set-l_in_if_work
var=foo
var=foo
```

Reply:
What happens is that `set` does not _modify_ $status.

So if you do

```fish
somecommand
if set
```

then that `if` still has the $status of `somecommand` to check.

Just like

```fish
if set somevar (somecommand)
```

checks the $status of `somecommand`.

However, setting a variable to some constant does not set $status.

I.e. `set -l var "foo"` has no status, so $status stays whatever it was.

```fish
false
set -l var "foo"
echo $status # prints 1
true
set -l var "foo"
echo $status # prints 0
```